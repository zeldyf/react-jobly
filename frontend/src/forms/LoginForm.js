import React, { useState } from 'react';
import { useHistory } from 'react-router-dom';
import { Card, CardBody, Form, Label, Input, Button } from 'reactstrap';

/** User login form.
 *
 * Shows form and manages update to state on changes.
 * On submission:
 * - calls login function prop
 * - redirects to /companies route
 *
 * Routes -> LoginForm
 * Routed as /login
 */

const LoginForm = ({ login }) => {
  const history = useHistory();
  const INITIAL_STATE = {
    username: '',
    password: ''
  };
  const [formData, setFormData] = useState(INITIAL_STATE);
  const [formErrors, setFormErrors] = useState([]);

  console.debug(
    'LoginForm',
    'login=',
    typeof login,
    'formData=',
    formData,
    'formErrors=',
    formErrors
  );

  /** Update form fields */

  const handleChange = evt => {
    const { name, value } = evt.target;
    setFormData(formData => ({
      ...formData,
      [name]: value
    }));
  };

  const handleSubmit = async evt => {
    evt.preventDefault();
    try {
      let result = await login(formData);
      // makes a POST request to Api.js and adds corresponding data to matching category in db.json
      if (result.success) {
        // imperatively redirect to correct page and refresh to see new data
        history.push('/');
      } else {
        setFormErrors(result.errors);
      }
    } catch (err) {
      console.log(err);
    }
  };

  return (
    <div className="LoginForm col-md-5 offset-md-4 col-lg-4 offset-lg-4">
      <Card>
        <CardBody>
        <h1>Log In</h1>
          <Form onSubmit={handleSubmit}>
            <Label htmlFor="username">Username</Label>
            <Input
              name="username"
              type="text"
              placeholder="Username"
              value={formData.username}
              onChange={handleChange}
              required
            ></Input>
            <Label htmlFor="password">Password</Label>
            <Input
              name="password"
              type="text"
              placeholder="Password"
              value={formData.password}
              onChange={handleChange}
              required
            ></Input>
            <span className="NewItemForm-message">
              {formErrors ? <p>{formErrors}</p> : null}
            </span>
            <Button
              type="submit"
              className="btn btn-lg btn-block"
              color="primary"
            >
              Login
            </Button>
          </Form>
        </CardBody>
      </Card>
      <p className='mt-3'>To explore and test out the features, feel free to log in using the username 'testuser' and the password 'password'.</p>
    </div>
  );
};

export default LoginForm;
