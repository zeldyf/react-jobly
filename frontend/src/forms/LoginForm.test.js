import React from 'react';
import { render } from '@testing-library/react';
import LoginForm from './LoginForm';

test('renders LoginForm without crashing', () => {
  render(<LoginForm />);
});

test('renders LoginForm correctly', () => {
  const { container } = render(<LoginForm />);
  expect(container).toMatchSnapshot();
});