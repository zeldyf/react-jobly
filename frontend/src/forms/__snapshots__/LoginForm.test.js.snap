// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LoginForm correctly 1`] = `
<div>
  <div
    class="LoginForm col-md-5 offset-md-4 col-lg-4 offset-lg-4"
  >
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <h1>
          Log In
        </h1>
        <form
          class=""
        >
          <label
            class=""
            for="username"
          >
            Username
          </label>
          <input
            class="form-control"
            name="username"
            placeholder="Username"
            required=""
            type="text"
            value=""
          />
          <label
            class=""
            for="password"
          >
            Password
          </label>
          <input
            class="form-control"
            name="password"
            placeholder="Password"
            required=""
            type="text"
            value=""
          />
          <span
            class="NewItemForm-message"
          >
            <p />
          </span>
          <button
            class="btn btn-lg btn-block btn btn-primary"
            type="submit"
          >
            Login
          </button>
        </form>
      </div>
    </div>
    <p
      class="mt-3"
    >
      To explore and test out the features, feel free to log in using the username 'testuser' and the password 'password'.
    </p>
  </div>
</div>
`;
